yarn-project/foundation/src/eth-address/index.ts

```diff
@@ -74,6 +74,27 @@ export class EthAddress {
     }
   }
 
+  /**
+   * Checks if the given value is an EthAddress instance even if the prototype chain is incorrect
+   * @param value - A value to check if it is an EthAddress instance.
+   * @returns Whether the value is an EthAddress instance.
+   */
+  public static isEthAddress(value: any): value is EthAddress {
+    if (!value || typeof value !== 'object') {
+      return false;
+    }
+
+    if (value instanceof EthAddress) {
+      return true;
+    }
+
+    if ('buffer' in value && Buffer.isBuffer(value.buffer) && value.buffer.length === EthAddress.SIZE_IN_BYTES) {
```

Comment:
nit: might be safer here to try/catch `fromBuffer`. this is making assumptions about the validity checks in the constructor.

Reply:
If I understand correctly, would it look something like:
```ts
try {
  if ('toBuffer' in value && EthAddress.fromBuffer(value.toBuffer()) {
    return true;
  }
} catch {
  return false;
}
```
I think that will work and be safe but we'd be creating objects and discarding them immediately.

Reply:
That is what I had meant, but perhaps the best approach is to extract the validation code to a function that both this typeguard and the constructor use?